packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -74,25 +77,50 @@ class Datalist extends Component {
 	}
 
 	getTitleMap() {
```

Comment:
making this function pure or calling a pure function doing the whole job, will make it lot easier to properly test.
I think it is important since this function does a lot more than before with many more execution path.